spikeinterface/core/waveform_extractor.py

```diff
@@ -245,6 +245,11 @@ def create(cls, recording, sorting, folder, mode="folder", remove_if_exists=Fals
         properties_to_attrs = deepcopy(recording._properties)
         if 'contact_vector' in properties_to_attrs:
             del properties_to_attrs['contact_vector']
+        if has_exceeding_spikes(recording, sorting):
```

Comment:
This is a high cost because it does a to vector.
But Iplan to refactor waveforms extractor using only vector version.
So lets go for this for now.

Reply:
it takes 1ms

Reply:
I do not believe you of course.
For enormous sorting also ? Or only for a small 10s test file ?

Reply:
>1h long neuropixels output ;) 

I think we have to have an internal caching of the spike vector in the sorting object, but that's another story ;)